src/escpos/escpos.py

```diff
@@ -396,6 +399,26 @@ def barcode(self, code, bc, height=64, width=3, pos="BELOW", font="A",
         if function_type.upper() == "A":
             self._raw(NUL)
 
+    def soft_barcode(self, barcode_type, data, module_height=5, module_width=0.2, text_distance=1):
+        image_writer = ImageWriter()
+
+        if barcode_type not in barcode.PROVIDED_BARCODES:
+            raise BarcodeTypeError(
+                'Barcode type {} not supported by software barcode renderer'
+                .format(barcode_type))
+
+        barcode_class = barcode.get_barcode_class(barcode_type)
+        my_code = barcode_class(data, writer=image_writer)
+
+        my_code.write("/dev/null", {
+            'module_height': module_height,
+            'module_width': module_width,
+            'text_distance': text_distance
+        })
+
+        image = my_code.writer._image
```

Comment:
We should patch pyBarcode upstream in order to add a property to the writer that returns the Pillow image. Since upstream seems dead maybe this is an opportunity to fork or find forks of pyBarcode that are more active.